src/main/java/org/openshift/jenkins/plugins/openshiftlogin/OpenShiftOAuth2SecurityRealm.java

```diff
@@ -605,16 +612,35 @@ public HttpResponse onSuccess(String authorizationCode) {
         };
     }
     
-    public void updateAuthorizationStrategy(OAuthSession oauth) throws IOException, GeneralSecurityException {
+    public UsernamePasswordAuthenticationToken updateAuthorizationStrategy(Credential credential) throws IOException, GeneralSecurityException {
     	populateDefaults();
-        OpenShiftUserInfo info = getOpenShiftUserInfo(oauth.getCredential(), transport);
-        Set<String> allowedRoles = postSAR(oauth.getCredential(), transport);
+        OpenShiftUserInfo info = getOpenShiftUserInfo(credential, transport);
+        Set<String> allowedRoles = postSAR(credential, transport);
         GrantedAuthority[] authorities = new GrantedAuthority[] {SecurityRealm.AUTHENTICATED_AUTHORITY};
+                
+        // we append the role suffix to the name stored into Jenkins, since a given user is able to log in at varying scope/permission
+        // levels in openshift; however, for now, we make sure the display name for Jenkins does not include this suffix
+        String suffix = "";
+        if (allowedRoles.contains("admin"))
+            suffix = "-admin";
+        else if (allowedRoles.contains("edit"))
+            suffix = "-edit";
+        else if (allowedRoles.contains("view"))
+            suffix = "view";
```

Comment:
Perhaps I should add comments to better clarify, but independent of browser or curl, a given user as defined in openshift can have up to 3 entries (each a unique permission level) into jenkin's auth matrix, where none of the keys for those entries is the user name defined in openshift.  The base user name used for the key is not necessarily what they might have supplied on the login, but whatever oauth self check on the token reports as the user name (where that token is supplied in the curl case, and is provided by the oauth server in the browser case).  

Then, that base user name is never stored in the Jenkins auth matrix.  Only <base user name>-admin, <base user name>-edit, <base user name>-view.

Then, based on the self-sar results, you get one of those 3 permission levels.